### DISCUSSION TITLE
Indexing provide incorrect resutls

### DISCUSSION THREAD

**Discussion by almogsegal** [photoprism/photoprism#3422] [Q&A]
Hi, 

First, I'd like to thank you for your hard work and the great application. 

Second, I been experimenting Photoprism as a replacement for Google Photos and found some issues with the latest container.
I've used the "Import" option to import and organized my photos. All the data seems to be imported, organized and renamed as expected but the indexing is incorrect. My photos library contains ~15748 items but "only" 14k of them were displayed in Photoprism. In addition, under the videos section it said that have over 600 videos while I have ~364 (it looks like it couldn't recognize that some of them are Live Photos). Rolling back to 221118-jammy container and performing a complete indexing solved all the issues. Same happened with my wife's library only that for her Photoprism found ~8k items while she has over 22k. Same as with my library, rolling back to 221118-jammy fixed the issue. I'm now trying to use latest again with hope that complete indexing wouldn't change the current status (in progress: currently it seems to have indexed some of the Live Photos as videos again). 

Thanks again,
Almog

**Comment by lastzero** [comment]
I've converted this to a discussion because there seem to be several potential issues (that no one else has reported to us), while you provide no information on how to reproduce them. Some of this could be a misunderstanding of how our software works, some of it could relate to feature requests that simply haven't been implemented yet. As for the counts, I suggest you take a look at our documentation to see if it explains things (we know PhotoPrism works differently than Google Photos e.g. supports stacks that Google may not have):

https://docs.photoprism.app/user-guide/faq/#media-library

**Reply by almogsegal** [reply]
That's correct. I been wondering if I should file a bug for a while now because I cannot pinpoint a specific photo/video that weren't indexed and give it to you as a reproducer. 
The weird part that caught me is that downgrading to a previous version got it all solved.
To be honest, I read the documentation quite a lot and couldn't find something that helped with that specific issue. I'll continue to look for it. 
Thanks.

**Reply by lastzero** [reply]
👉 https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

**Reply by almogsegal** [reply]
I've seen this one as well :)

**Reply by lastzero** [reply]
Maybe a full rescan would have solved it with our latest release too? Also note that recent versions don't count files under review, see the release notes. This could explain it: 
-  https://github.com/photoprism/photoprism/issues/3164

**Reply by almogsegal** [reply]
Latest release is the one that causes the issue. Only 221118-jammy shows everything fine. Nothing is under review. Did you mean preview?

**Reply by lastzero** [reply]
See the linked issue for when exactly the change was implemented. This was earlier this year. So it's all the 2023 releases including our preview build.

**Reply by almogsegal** [reply]
So that's probably unrelated. I have nothing under review.

**Reply by lastzero** [reply]
Could it be that some files have the wrong extension? Google sometimes does this and we now are stricter with this to improve indexing performance.

**Reply by almogsegal** [reply]
I don't think so. Also, why would 221118 work fine then?

**Reply by almogsegal** [reply]
There are two issues here:

1. Not all the data is indexed with latest build. Works fine with 221118. 
2. Many Apple Live Photos are not recognized as such and PP recognizes them as videos (I see many of them of length 2-4 seconds under Videos and not Live Photos). Also works fine with 221118.

I think that the two of them might be related as PP cannot find some photos and therefore it doesn't understand that these 2-4 seconds videos should be linked to a photo.

**Reply by lastzero** [reply]
Aren't there any error or warnings in the logs who could give you a hint? As I've said, files night be skipped or ignored if the mime type doesn't match with the file extension. If you have been using Google Photos so far and did an export, there's a good chance this might be the case for some of your pictures. If so, please contact Google to file a bug report there.

**Reply by almogsegal** [reply]
I didn't see a specific error. Do I need to re-index the library with the latest version to reproduce the errors or is there a place with old logs? 
I don't think the the issue is with the Google export as everything works fine with PP 221118...

**Reply by lastzero** [reply]
The indexing performance has been improved with our latest releases, in part due to better file type validation. This causes, for example, JPEG files with a PNG extension to be ignored. See the Release Notes. It can therefore be related. Since the files now seem to be indexed, there's a chance it works for you if you upgrade again. Since I've been exporting files from Google in the past, I know for a fact that it can include wrong file extensions - unless they have fixed it somehow.

**Reply by lastzero** [reply]
Old and new errors and warnings can be found here:

https://demo.photoprism.app/library/errors

**Reply by almogsegal** [reply]
There's a single error that repeats over and over in the logs:
```
import: Input file is not an HEIF/AVIF file in 2019/10/20191005_152746_643E7715.heic (convert to jpeg) 
```
Do you think this might be the issue? Was there a functionality in 221118 that handled these cases? 

By the way, when it says "convert to jpeg", does it mean that it modifies the file in my originals or it asks me to do so?

**Reply by lastzero** [reply]
I can tell you this if you send us a file sample (but note this is a service for members and I'm currently on vacation): https://www.photoprism.app/contact

In the old release, a mime type detection was performed before checking the extension. This takes time. Now the extension is checked first. In any event, extension and type MUST match. If not, it's a bug in the Google export, not our indexer.

Your originals won't get modified as described in our documentation:

https://docs.photoprism.app/getting-started/faq/#what-media-file-types-are-supported

**Reply by almogsegal** [reply]
I think you were right about the wrong extension.
I used exiftool to validate the photos (`exiftool -validate -error -warning -r -a`) and got 1221 wrong file extension (exiftool warning is: File has wrong extension (should be JPG, not HEIC)) and over 9k when I checked my wife's library. Calculating the numbers (on my wife's library for example) 21566 - 9389 - ~4k of review files gets me to the ~8k photos that were found...

The question is, how does the PP extension validation process goes? does it compare image extension to the Exif FileExtension field?

Please don't spend your vacation on answering here. There's no rush.

**Reply by lastzero** [reply]
I'll forget it otherwise as we get new questions every day 😉

PhotoPrism has its own mimetype detection, we don't use exiftool for this. To make sure it works, we have an extensive test suite with sample files for different formats. As running exiftool also takes time, we only do this if necessary.

What might have happened is that Google replaced your HEIC files with JPEGs without keeping the originals. Then, they thought they can fool you by keeping the original extension? Or their software doesn't support renaming files? Or maybe there just was no budget for this? 🤔

**Reply by almogsegal** [reply]
Haha I know that feeling :)

I see. I'm wondering if I should rename these files to JPEG and get it over with. I'll try to file a bug to Google anyway...

Yes, that's probably the case...

Thank you very much for your help! Enjoy your vacation :)
